native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/NativeImageExecutableLocator.java

```diff
@@ -91,29 +91,32 @@ public static File findNativeImageExecutable(Property<JavaLauncher> javaLauncher
             executablePath = metadata.getInstallationPath().file("bin/" + NATIVE_IMAGE_EXE).getAsFile();
         }
 
-        try {
-            if (!executablePath.exists()) {
-                logger.log("Native Image executable wasn't found. We will now try to download it. ");
-                File graalVmHomeGuess = executablePath.getParentFile();
-
-                File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
-                if (!guPath.exists()) {
-                    throw new GradleException("'" + GU_EXE + "' at '" + guPath + "' tool wasn't found. This probably means that JDK at isn't a GraalVM distribution.");
-                }
+        File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
+        if (guPath.exists() && !executablePath.exists()) {
+            logger.log("Native Image executable wasn't found. We will now try to download it. ");
+            File graalVmHomeGuess = executablePath.getParentFile();
+
+            File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
```

Comment:
Unless something is wrong with the diff, it seems you are re-declaring `guPath` here,and then also check for it's existence again.